### PR TITLE
[BugFix] Fix the issue of csv trimspace

### LINKED ISSUE #19196: [csv para] stream load and set trim_space =true， enclose=", result is incorrect
### Steps to reproduce the behavior (Required)
1. CREATE TABLE '...'
```
create database stream_load_test_db_eba7a9dc_bd57_11ed_95d6_00163e0e550b;
use stream_load_test_db_eba7a9dc_bd57_11ed_95d6_00163e0e550b;
 CREATE TABLE `duplicate_table_with_null` (
                k0 int,v1 date,v2 datetime,v3 char(20),v4 varchar(40),v5 string
                ) ENGINE=OLAP
            DUPLICATE KEY(k0)
            COMMENT "OLAP"
            DISTRIBUTED BY HASH(k0) BUCKETS 3
            PROPERTIES (
                "replication_num" = "3",
                "storage_format" = "v2"
            ) ;
curl --location-trusted -u root: -T /home/disk4/huoyue/pull_StarRocksTest2/StarRocksTest/lib/../common/data/para_trim_space_enclose.csv -XPUT -H label:stream_load_1678242083195 -H trim_space:true -H enclose:"\"" -H column_separator:, http://xxx.xx.xx.xxx:8234/api/stream_load_test_db_eba7a9dc_bd57_11ed_95d6_00163e0e550b/duplicate_table_with_null/_stream_load
```
data
```
 1 , 9999-12-31 , 9999-12-31 23:59:59 , "beijintgaergertte"  ,  "haindian'sdvgerwwge"  ,  "asrew    er:""hgeth "
 2 , 0001-01-01 , 0001-01-01 00:00:01 , " beij  ing "  ,  " haidtnr&*_#@#$)!_ian "  ,  " saefwd *""* "
 3 , 2020-01-01 , 2020-01-01 00:00:01 , "beij  ting"  ,  "haidi""tnr&*_#@#$)!_ian"  ,  "saefwd *""* "
```

### Real behavior (Required)
image

### StarRocks version (Required)
 - You can get the StarRocks version by executing SQL `select current_version()`
mysql> select current_version();
+----------------------------+
| current_version()          |
+----------------------------+
| BRANCH-3.0-RELEASE d659a54 |
+----------------------------+
1 row in set (0.01 sec)

### PR DESCRIPTION
**Pull request by zaorangyang**
## What type of PR is this：
- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #19196

## Problem Summary(Required) ：

This PR change the logic of trim space. If you turn trimspace on, leading and trailing spaces for fields wrapped in enclose characters will be ignored.
For example, assuming the field delimiter is | and enclose character = '"':
```
|"Love StarRocks"|
|" Love StarRocks "|
| "Love StarRocks" |
```
becomes:
```
|"Love StarRocks"|
|" Love StarRocks "|
|"Love StarRocks"|
```

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto backported to target branch
  - [x] 3.0
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3

### CODE REVIEW

File: be/src/formats/csv/csv_reader.cpp [hunk 1]
```diff
@@ -325,6 +328,16 @@ Status CSVReader::more_rows() {
             }
 
         case ORDINARY:
+            //
```
Comment (chaoyli) [inline be/src/formats/csv/csv_reader.cpp:331]: This line is non-sense.

File: be/src/formats/csv/csv_reader.cpp [hunk 2]
```diff
@@ -325,6 +328,16 @@ Status CSVReader::more_rows() {
...
+            if (UNLIKELY(_parse_options.trim_space)) {
```
Comment (chaoyli) [inline be/src/formats/csv/csv_reader.cpp:332]: It's not efficient. You can find the last space in one loop if there are multiple spaces.

Comment (zaorangyang) [reply be/src/formats/csv/csv_reader.cpp:332]: In fact, we just only need target the first whitespace.

**Review by chaoyli — DISMISSED** [review]

**Review by meegoo — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Review by chaoyli — APPROVED** [review]

**Comment by zaorangyang** [comment]
run starrocks_be_unittest

**Review by meegoo — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.0

**Pull request merged**